### ISSUE TITLE
[optimizer] feature/port idea - making this part of Varnish

### ISSUE THREAD

**Issue by QES** [ampproject/amp-toolbox#42]
Hi,

I just got a Varnish error on the ampproject site (https://www.ampproject.org/) and so I can see this is something the team use :)

I'm a great fan of Varnish and wondered if anyone had thought of creating a version of the amp-toolbox to sit as part of the amp-toolbox project it seems like a perfect place to include such a tool as it would take all the complexity out of implementation and provide a way to make it transparent to the back end and the front end user.

**Comment by sebastianbenz** [comment]
Yes, this would make a lot of sense! Unfortunately, we don't have the capacity to write specific CDN/tool integrations. One goal of amp-toolbox  is to enable others to write such integrations though (hint hint). 

Closing this issue for now as out-of-scope.

**Comment by QES** [comment]
Thanks @sebastianbenz if I could I would but my skills aren't there quite yet.

If I understand things Varnish is written in C so it would require porting the current amp-toolbox before starting.

Is there a high level view of what amp-toolbox is doing to the pages?
